internal/version/version_test.go

```diff
@@ -0,0 +1,332 @@
+			v1 := Version{Major: 9, Minor: 0, Patch: 0, Build: 0}
+			v2 := Version{Major: 8, Minor: 0, Patch: 0, Build: 0}
+			Expect(v1.IsGreaterOrEqual(&v2)).To(BeTrue())
+		})
+
+		It("should return true when versions are equal", func() {
+			v1 := Version{Major: 8, Minor: 0, Patch: 1, Build: 0}
+			v2 := Version{Major: 8, Minor: 0, Patch: 1, Build: 0}
+			Expect(v1.IsGreaterOrEqual(&v2)).To(BeTrue())
+		})
+
+		It("should return false when version is smaller", func() {
+			v1 := Version{Major: 7, Minor: 0, Patch: 0, Build: 0}
+			v2 := Version{Major: 8, Minor: 0, Patch: 0, Build: 0}
+			Expect(v1.IsGreaterOrEqual(&v2)).To(BeFalse())
+		})
+	})
+
+	Describe("IsSmallerOrEqual", func() {
+		It("should return true when version is smaller", func() {
+			v1 := Version{Major: 7, Minor: 0, Patch: 0, Build: 0}
+			v2 := Version{Major: 8, Minor: 0, Patch: 0, Build: 0}
+			Expect(v1.IsSmallerOrEqual(&v2)).To(BeTrue())
+		})
+
+		It("should return true when versions are equal", func() {
+			v1 := Version{Major: 8, Minor: 0, Patch: 1, Build: 0}
+			v2 := Version{Major: 8, Minor: 0, Patch: 1, Build: 0}
+			Expect(v1.IsSmallerOrEqual(&v2)).To(BeTrue())
+		})
+
+		It("should return false when version is greater", func() {
+			v1 := Version{Major: 9, Minor: 0, Patch: 0, Build: 0}
+			v2 := Version{Major: 8, Minor: 0, Patch: 0, Build: 0}
+			Expect(v1.IsSmallerOrEqual(&v2)).To(BeFalse())
+		})
+	})
+
+	Describe("Real-world scenarios", func() {
+		Context("parsing and comparing version strings", func() {
+			DescribeTable("should compare versions correctly",
+				func(v1Str, v2Str, expectedResult string) {
+					v1, err := NewVersion(v1Str)
+					Expect(err).ToNot(HaveOccurred())
+
+					v2, err := NewVersion(v2Str)
+					Expect(err).ToNot(HaveOccurred())
+
+					switch expectedResult {
+					case "greater":
+						Expect(v1.IsGreater(v2)).To(BeTrue())
+					case "smaller":
+						Expect(v1.IsSmaller(v2)).To(BeTrue())
+					case "equal":
+						Expect(v1.IsEqual(v2)).To(BeTrue())
+					}
+				},
+				Entry("release vs patch", "8.1.0.0", "8.0.1.0", "greater"),
+				Entry("major upgrade", "9.0.0.0", "8.9.9.9", "greater"),
+				Entry("build difference", "8.0.1.1", "8.0.1.0", "greater"),
+				Entry("same versions", "8.0.1.0", "8.0.1.0", "equal"),
+				Entry("partial vs full", "8.1", "8.0.5.10", "greater"),
+			)
+		})
+	})
+
+	Describe("Edge cases", func() {
+		It("should handle very large version numbers", func() {
+			version, err := NewVersion("999999.999999.999999.999999")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(*version).To(Equal(Version{Major: 999999, Minor: 999999, Patch: 999999, Build: 999999}))
+		})
+
+		It("should handle single digit zero version", func() {
+			version, err := NewVersion("0")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(*version).To(Equal(Version{Major: 0, Minor: 0, Patch: 0, Build: 0}))
+		})
+
+		It("should reject versions with extra text", func() {
+			_, err := NewVersion("8.0.1.0-beta")
+			Expect(err).To(HaveOccurred())
```

Comment:
This should be acceptable and equal to `8.0.1.0`. Of course that's incorrect in general case, but ok for us.